src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,16 @@
+import type { Dispatch } from 'react';
+import { useState } from 'react';
+
+export const useLocalStorage = <T>(key: string, initialType: object[] | object) => {
+  const [localStorageData, setLocalStorageData] = useState<T>(
+    JSON.parse(localStorage.getItem(key) ?? `${JSON.stringify(initialType)}`),
+  );
+
+  const internalSetLocalStorageData: Dispatch<T> = (nextLocalStorageData) => {
+    localStorage.setItem(key, JSON.stringify(nextLocalStorageData));
+
+    setLocalStorageData(nextLocalStorageData);
+  };
+
+  return { localStorageData, internalSetLocalStorageData };
+};
```

Comment:
```suggestion
import type { Dispatch } from 'react';
import { useState } from 'react';

export const useLocalStorage = <T extends object | object[]>(key: string, defaultValue: T) => {
  const [localStorageData, setLocalStorageData] = useState<T>(
    JSON.parse(localStorage.getItem(key) ?? `${JSON.stringify(defaultValue)}`),
  );

  const internalSetLocalStorageData: Dispatch<T> = (nextLocalStorageData) => {
    localStorage.setItem(key, JSON.stringify(nextLocalStorageData));

    setLocalStorageData(nextLocalStorageData);
  };

  return { localStorageData, internalSetLocalStorageData };
};
```
더 명확한 타입과 적절한 네이밍을 제안해봅니다 😁

Reply:
defaultValue가 더 정확한 네이밍이네요! 그리고 제네릭 사용이 아직 미흡했는데 이렇게 하는게 더 명확한 타입이겠군요! 변경했습니다!